SwiftCompilerSources/Sources/SIL/ApplySite.swift

```diff
@@ -108,6 +108,14 @@ public protocol ApplySite : Instruction {
 extension ApplySite {
   public var callee: Value { operands[ApplyOperandConventions.calleeIndex].value }
 
+  public var hasSubstitutions: Bool {
+    return substitutionMap.hasAnySubstitutableParams
+  }
+
+  public var canOptimize: Bool {
```

Comment:
Please choose another name for this. What kind of optimizations are not possible? I think it makes sense to move this as a private utility to the AutodiffClosureSpecializer. It's very specific to this optimization.